lib/commands/electron.js

```diff
@@ -1,127 +1,121 @@
 'use strict'
 
-const spawn = require('child_process').spawn
-const chalk = require('chalk')
 const RSVP = require('rsvp')
-const debugServer = require('../helpers/debug-server')
-const safePath = require('../helpers/safe-path')
-const getRemoteDebugSocketScript = require('../helpers/remote-debug-script')
-const os = require('os')
 
-const Promise = RSVP.Promise
+const Command = require('ember-electron/lib/commands/-command')
+const InspectorServer = require('ember-electron/lib/utils/inspector-server')
+const Logger = require('ember-electron/lib/utils/logger')
+const WatchedBuild = require('ember-electron/lib/utils/watched-build')
+const start = require('electron-forge/dist/api/start').default
 
-module.exports = {
+const { Promise } = RSVP
+
+module.exports = Command.extend({
   name: 'electron',
   description: 'Builds your app and launches Electron',
 
-  init: function () {
-    process.env.EMBER_CLI_ELECTRON = true
-
-    if (this._super && this._super.init) {
-      this._super.init.apply(this, arguments)
-    }
-  },
-
   availableOptions: [{
     name: 'environment',
     type: String,
     default: 'development',
-    aliases: ['e', {
-      dev: 'development'
-    }, {
-      prod: 'production'
-    }]
+    aliases: [
+      'e',
+      { dev: 'development' },
+      { prod: 'production' }
+    ]
   }, {
     name: 'output-path',
     type: String,
     default: 'dist/',
     aliases: ['o']
+  }, {
+    name: 'verbose',
+    type: Boolean,
+    default: false,
+    aliases: ['v']
   }],
 
-  buildWatch: function (options) {
-    const ui = this.ui
-    const Watcher = require('ember-cli/lib/models/watcher')
-    const Builder = require('ember-cli/lib/models/builder')
-
-    ui.startProgress(chalk.green('Building'), chalk.green('.'))
-
-    const watcher = new Watcher({
-      ui: ui,
-      builder: new Builder({
-        ui: ui,
-        project: this.project,
-        environment: options.environment,
-        outputPath: options.outputPath
-      }),
-      analytics: this.analytics,
-      options: options
-    })
+  run (options) {
+    let logger = new Logger(this)
 
-    return watcher
-  },
+    return this._buildAndWatch(options, logger)
+      .then(() => this._runInspectorServer(options, logger))
+      .then(() => {
+        logger.message('Starting Electron...')
+        logger.line()
 
-  runInspectorServer: function (options) {
-    if (options.environment !== 'development') {
-      return
-    }
+        return start()
```

Comment:
@anulman Something is baffling me here as I'm trying to figure out how to get `ember electron:test` running through electron forge, rather than requiring electron to be installed in the host app. This `start()` call doesn't have any options, and I don't see any chdirs, so how does electron-forge know to run against the output of the build command?